Sources/Bridges/OTelSwiftLog/LogHandler.swift

```diff
@@ -0,0 +1,192 @@
+        }
+
+        // Convert metadata from the struct property to AttributeValue and merge it with otelattributes
+        let structMetadata = convertMetadata(self.metadata)
+        otelattributes.merge(structMetadata) { _, new in new }
+
+        // Build the log record and emit it
+        let event = self.logger.logRecordBuilder().setSeverity(convertSeverity(level: level))
+        .setBody(AttributeValue.string(message.description))
+        .setAttributes(otelattributes)
+      
+      if let context = OpenTelemetry.instance.contextProvider.activeSpan?.context {
+          _ = event.setSpanContext(context)
+      }
+        event.emit()
+
+    }
+
+
+
+}
+
+extension LoggerBuilder {
+    @discardableResult
+    func configure(with instrumentationScope: InstrumentationScope) -> LoggerBuilder {
+        var builder = self
+        
+        if let version = instrumentationScope.version {
+            builder = builder.setInstrumentationVersion(version)
+        }
+        
+        if let eventDomain = instrumentationScope.eventDomain {
+            builder = builder.setEventDomain(eventDomain)
+        }
+        
+        if let schemaUrl = instrumentationScope.schemaUrl {
+            builder = builder.setSchemaUrl(schemaUrl)
+        }
+        
+        if let includeTraceContext = instrumentationScope.includeTraceContext {
+            builder = builder.setIncludeTraceContext(includeTraceContext)
+        }
+        
+        if let attributes = instrumentationScope.attributes {
+            builder = builder.setAttributes(attributes)
+        }
+        
+        return builder
+    }
+}
+
+func convertMetadata(_ metadata: Logging.Logger.Metadata) -> [String: AttributeValue] {
+    var convertedAttributes: [String: AttributeValue] = [:]
+
+    // Iterate over each key-value pair in the metadata dictionary
+    for (key, value) in metadata {
+        // Convert each value to AttributeValue
+        let attributeValue = convertToAttributeValue(value)
+
+        // Store the converted value with its corresponding key in the attributes dictionary
+        convertedAttributes[key] = attributeValue
+    }
+
+    return convertedAttributes
+}
+
+// Function to recursively convert nested dictionaries to AttributeValue
+func convertToAttributeValue(_ value: Logging.Logger.Metadata.Value) -> AttributeValue? {
+    switch value {
+    case .dictionary(let nestedDictionary):
+        // If value is a nested dictionary, recursively convert it
+        var nestedAttributes: [String: AttributeValue] = [:]
+        for (nestedKey, nestedValue) in nestedDictionary {
+            nestedAttributes[nestedKey] = convertToAttributeValue(nestedValue)
+        }
+        return AttributeValue.set(AttributeSet(labels: nestedAttributes))
+    case .array(let nestedArray):
+        // If value is a nested array, recursively convert it
+        let nestedValues = nestedArray.map { convertToAttributeValue($0) }
```

Comment:
`ConvertToAttributeValue` returns a `AttributeValue`, thus the `nestedValues` variable is of the type `[AttributeValue]`. All the subsequent checks are done for arrays of literals, thus this will always return nil.